homeassistant/components/proximity/__init__.py

```diff
@@ -3,138 +3,31 @@
 from __future__ import annotations
 
 import logging
-from typing import cast
 
 import voluptuous as vol
 
-from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
-from homeassistant.const import (
-    CONF_DEVICES,
-    CONF_NAME,
-    CONF_UNIT_OF_MEASUREMENT,
-    CONF_ZONE,
-    STATE_UNKNOWN,
-    Platform,
-)
-from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN, HomeAssistant
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.event import (
     async_track_entity_registry_updated_event,
     async_track_state_change_event,
 )
-from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
-from homeassistant.helpers.typing import ConfigType
-from homeassistant.helpers.update_coordinator import CoordinatorEntity
-
-from .const import (
-    ATTR_DIR_OF_TRAVEL,
-    ATTR_DIST_TO,
-    ATTR_NEAREST,
-    CONF_IGNORED_ZONES,
-    CONF_TOLERANCE,
-    CONF_TRACKED_ENTITIES,
-    DEFAULT_PROXIMITY_ZONE,
-    DEFAULT_TOLERANCE,
-    DOMAIN,
-    UNITS,
-)
+
+from .const import CONF_TRACKED_ENTITIES, DOMAIN
 from .coordinator import ProximityConfigEntry, ProximityDataUpdateCoordinator
-from .helpers import entity_used_in
 
 _LOGGER = logging.getLogger(__name__)
 
-ZONE_SCHEMA = vol.Schema(
-    {
-        vol.Optional(CONF_ZONE, default=DEFAULT_PROXIMITY_ZONE): cv.string,
-        vol.Optional(CONF_DEVICES, default=[]): vol.All(cv.ensure_list, [cv.entity_id]),
-        vol.Optional(CONF_IGNORED_ZONES, default=[]): vol.All(
-            cv.ensure_list, [cv.string]
-        ),
-        vol.Optional(CONF_TOLERANCE, default=DEFAULT_TOLERANCE): cv.positive_int,
-        vol.Optional(CONF_UNIT_OF_MEASUREMENT): vol.All(cv.string, vol.In(UNITS)),
-    }
-)
-
 CONFIG_SCHEMA = vol.Schema(
     vol.All(
         cv.deprecated(DOMAIN),
-        {DOMAIN: cv.schema_with_slug_keys(ZONE_SCHEMA)},
     ),
     extra=vol.ALLOW_EXTRA,
 )
```

Comment:
```suggestion
```

Reply:
AFAIK we need to keep the CONFIG_SCHEME, but mark it as removed -> 0226fb7

Reply:
AFAIK, the config schema is only needed if we have a `(async_)setup` function, which is also deleted with this PR. Therefore, we can remove it.

If there would be a `(async_)setup` function we should use [`cv.config_entry_only_config_schema`](https://github.com/home-assistant/core/blob/ef237a8431752655423c48b99f626c42ec60e212/homeassistant/helpers/config_validation.py#L1168) as the function docs says:
```
Return a config schema which logs if attempted to setup from YAML.

Use this when an integration's __init__.py defines setup or async_setup
but setup from yaml is not supported.
```